document/src/vespa/document/datatype/arraydatatype.h

```diff
@@ -12,21 +12,16 @@
 namespace document {
 
 class ArrayDataType final : public CollectionDataType {
-protected:
-    // Protected to help you avoid calling the copy constructor when
-    // you think you're calling the regular constructor with a nested
-    // ArrayDataType.
-    ArrayDataType(const ArrayDataType &o) : CollectionDataType(o) {}
-
 public:
     explicit ArrayDataType(const DataType &nestedType);
+    ArrayDataType(const ArrayDataType &o) = delete;
+    ArrayDataType &operator=(const ArrayDataType &rhs) = delete;
```

Comment:
Consider explicitly deleting move constructor and move assignment too.

Reply:
I will do another pass cleaning up the todos and also delete the move constructor/operator.
But for they are deleted implicit by the explicit destructor.